### DISCUSSION TITLE
Decentralized/modular routes for browser router?

### DISCUSSION THREAD

**Discussion by joeattardi** [remix-run/react-router#12009] [Q&A]
My application is broken up into feature modules which are loaded when the app starts. We have a lot of routes. Is there a way to, say, have each feature module define its own routes, then combine them all into the single app-wide router? 

Basically I am looking for a strategy to have a module register its own routes somehow with the router as the app is loaded.

**Comment by sergiodxa** [comment]
You could export the list of routes of a feature as an array and then in a centralized place combine them on the final list you pass to `createBrowserRouter`.

**Reply by jrestall** [reply]
This is the way. We have each feature module export a routes array and combine them all in the main app.
